monitoring/monitoring/monitoring.go

```diff
@@ -304,27 +304,49 @@ func (c *Container) renderRules() (*promRulesFile, error) {
 						continue
 					}
 
-					// The alertQuery must contribute a query that returns true when it should be firing.
-					var alertQuery string
+					var (
+						// Wrap the query in `max()` or `min()` so that if there are multiple series (e.g. per-container)
+						// they get "flattened" into a single metric. The `aggregator` variable sets the required operator.
+						//
+						// We only support per-service alerts, not per-container/replica, and not doing so can cause issues.
+						// See https://github.com/sourcegraph/sourcegraph/issues/11571#issuecomment-654571953,
+						// https://github.com/sourcegraph/sourcegraph/issues/17599, and related pull requests.
+						aggregator string
+						// Comparator sets how a metric should be compared against a threshold
+						comparator string
+						// Threshold sets the value to be compared against
+						threshold float64
+					)
+
+					// Set values to build a query with
 					if a.greaterThan != nil {
-						comparator := ">="
+						aggregator = "min" // alert if the largest value is exceeds upper bound
```

Comment:
```suggestion
						aggregator = "max" // alert if the largest value is exceeds upper bound
```
Text says the largest

Reply:
oops you're right!